plutus-tx-plugin/test/Budget/allEmptyList.plc.golden

```diff
@@ -23,7 +30,103 @@ letrec
+                                         (lessThanEqualsInteger eta eta)
+                                         True
+                                         False)
+                                      {all dead. Ordering}
+                                      (/\dead -> LT)
+                                      (/\dead -> GT)
+                                      {all dead. dead})
+                                 {all dead. dead})
+                            (\(x : integer)
+                              (y : integer) ->
+                               ifThenElse
+                                 {Bool}
+                                 (lessThanInteger x y)
+                                 True
+                                 False)
+                            (\(x : integer)
+                              (y : integer) ->
+                               ifThenElse
+                                 {Bool}
+                                 (lessThanEqualsInteger x y)
+                                 True
+                                 False)
+                            (\(x : integer)
+                              (y : integer) ->
+                               ifThenElse
+                                 {Bool}
+                                 (lessThanEqualsInteger x y)
+                                 False
+                                 True)
+                            (\(x : integer)
+                              (y : integer) ->
+                               ifThenElse
+                                 {Bool}
+                                 (lessThanInteger x y)
+                                 False
+                                 True)
+                            (\(x : integer)
+                              (y : integer) ->
+                               Bool_match
+                                 (ifThenElse
+                                    {Bool}
+                                    (lessThanEqualsInteger x y)
+                                    True
+                                    False)
+                                 {all dead. integer}
+                                 (/\dead -> y)
+                                 (/\dead -> x)
+                                 {all dead. dead})
+                            (\(x : integer)
+                              (y : integer) ->
+                               Bool_match
+                                 (ifThenElse
+                                    {Bool}
+                                    (lessThanEqualsInteger x y)
+                                    True
+                                    False)
+                                 {all dead. integer}
+                                 (/\dead -> x)
+                                 (/\dead -> y)
+                                 {all dead. dead})
+                    in
+                    Ord_match
+                      {integer}
+                      v
```

Comment:
A context-based inlining heuristic would help here: inline if the thing you are inlining is a constructor application and the variable is the scrutinee of a pattern match!

Reply:
Also need case-of-known-constructor.